tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-negative.bal

```diff
@@ -433,3 +433,56 @@ function testAssigningKeyedToEmptyKeyedTbl() {
     _ = tbl2[111];
     tbl2[222] = {id: 222, firstName: "Melina", lastName: "Kodel"};
 }
+
+type Person1 record {
+    readonly int id;
+    string name;
+};
+
+function testMultipleKeys1() {
+    table<Person1> key(id) t1 = table [
+        {
+            id: 1,
+            name: "a"
+        },
+        {
+            id: 2,
+            name: "a"
+        }
+    ];
+
+    Person1? _ = t1[1, 2, 3]; // error
+}
+
+type Person2 record {
+    readonly int id;
+    readonly string name;
+};
+
+function testMultipleKeys2() {
+    table<Person2> key(id) t1 = table [
+        {
+            id: 1,
+            name: "a"
+        },
+        {
+            id: 2,
+            name: "a"
+        }
+    ];
+
+    Person2? _ = t1[1, "2", "3"]; // error
+
+    table<Person2> key(id) t2 = table [
+        {
+            id: 1,
+            name: "a"
+        },
+        {
+            id: 2,
+            name: "a"
+        }
+    ];
+
+    Person2? _ = t2[11, 12, 13, 14]; // error
+}
```

Comment:
Shall we add a test with key-type-constraint as well

Reply:
Added